frontend/src/Frontend/Transfer.hs

```diff
@@ -0,0 +1,235 @@
+            e <- case M.lookup "Chainweb-Next" headers of
+              Nothing -> pure never
+              Just next -> evaporateButtonOnClick next t
+            pure ()
+
+      pure ()
+
+nextButtonText :: Text
+nextButtonText = "Fetch more results..."
+
+
+fetchButton :: DomBuilder t m => Text -> m (Event t ())
+fetchButton s = do
+  let ourAttrs = mconcat [ "type" =: "button", "class" =: "ui blue button", "style" =: "margin: auto" ]
+  (e, _) <- elAttr' "button" ourAttrs $ text s
+  return $ domEvent Click e
+
+disabledButton :: DomBuilder t m => Text -> m (Event t ())
+disabledButton s = do
+  let ourAttrs = mconcat [ "type" =: "button", "class" =: "ui disabled button", "style" =: "margin: auto" ]
+  (e,_) <- elAttr' "button" ourAttrs $ text s
+  return $ domEvent Click e
+
+evaporateButtonOnClick
+  :: MonadFix m
+  => Reflex t
+  => DomBuilder t m
+  => MonadHold t m
+  => PostBuild t m
+  => Text -> Event t (Either TransferError (Maybe Text)) -> m (Event t Text)
+evaporateButtonOnClick token t = mdo
+  let something = leftmost [(inlineLoader "Fetching more rows..." >> pure never) <$ click, v]
+      v = t <&> \case 
+        Right (Just newToken) -> (const newToken) <$$> fetchButton nextButtonText
+        Right Nothing -> pure never
+        Left (NonHTTP200 status) -> disabledButton ("Non 200 HTTP Status: " <> T.pack (show status)) >> pure never
+        Left MissingHeader -> disabledButton "Missing Chainweb-Next Header"  >> pure never
+        Left BadResponse -> disabledButton "Bad response" >> pure never
+        Left NoResponseText -> disabledButton "Missing response in HTTP Request" >> pure never
+  beenClicked <- holdDyn ((const token) <$$> fetchButton nextButtonText) something
+  clickNow <- networkView beenClicked
+  click <- switchHold never clickNow
+  pure click
+
+type TransferXHR = Maybe Int -> Maybe Int -> Maybe Text -> Either String (XhrRequest ())
+
+data TransferError = NonHTTP200 Word | MissingHeader | BadResponse | NoResponseText
+
+produceNewRowsOnToken 
+  :: (MonadApp r t m, MonadJSM (Performable m),
+  HasJSContext (Performable m), Prerender js t m,
+  RouteToUrl (R FrontendRoute) m, SetRoute t (R FrontendRoute) m, MonadIO m)
+  => TransferXHR -> NetId -> Text -> Text -> Maybe Integer -> Event t Text -> m (Event t (Either TransferError (m (), Text)))
+produceNewRowsOnToken mkXhr n token account chainid nextToken = do
+  result <- performRequestAsync $ fmap (\t -> either error id $ mkXhr Nothing Nothing (Just t)) nextToken -- TODO: Don't use error here
+  return $ result <&> \xhr -> if _xhrResponse_status xhr /= 200 
+     then Left $ NonHTTP200 (_xhrResponse_status xhr) 
+     else do
+         r <- note NoResponseText $ _xhrResponse_responseText xhr
+         details <- note BadResponse $ getAccountDetail r
+         let headers = _xhrResponse_headers xhr
+         let rowsToRender = forM_ details $ \detail -> el "tr" $ drawRow n token account chainid detail
```

Comment:
I think we should move this `drawRow` call out of `produceNewRowsOnToken`, because the `n`, `token`, `account` and `chainid` parameters aren't used anywhere else and this might just be an unnecessary indirection.